apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -41,11 +42,57 @@ export default class AnimationPickerBody extends React.Component {
   state = {
     searchQuery: '',
     categoryQuery: '',
-    currentPage: 0
+    currentPage: 0,
+    headerOffset: HEADER_FOOTER_OFFSET_PADDING,
+    categoryHeight: 0,
+    footerHeight: 0
   };
 
   componentDidMount() {
     this.multiSelectEnabled_ = experiments.isEnabled(experiments.MULTISELECT);
+    // Calculate the space taken up by the title and search bar. Title and search bar are visible in all states of
+    // animation picker body.
+    let headerOffset = HEADER_FOOTER_OFFSET_PADDING;
+
+    if (this.refs && this.refs.animationPicker_title) {
+      headerOffset += this.refs.animationPicker_title.clientHeight;
+    }
+
+    if (this.refs && this.refs.animationPicker_searchBar) {
+      headerOffset += this.refs.animationPicker_searchBar.clientHeight;
+    }
+
+    this.setState({headerOffset: headerOffset});
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    // Calculate the space take up by the category breadcrumbs and footer 'done' button. Only visible when search
+    // is engaged or category is selected.
+
+    // Determine if category or search changed.
+    if (
+      prevState.searchQuery !== this.state.searchQuery ||
+      prevState.categoryQuery !== this.state.categoryQuery
+    ) {
+      // Determine if category or search state is engaged.
+      if (this.state.searchQuery !== '' || this.state.categoryQuery !== '') {
+        if (this.refs && this.refs.animationPicker_categories) {
```

Comment:
i think this code will be removed, but anytime you notice a guard like this, we should be using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) for readability